src/prefect/infrastructure/provisioners/cloud_run.py

```diff
@@ -191,44 +195,93 @@ async def _create_gcp_credentials_block(
                                     - Service Account User
                                     - Cloud Run Developer
-                            - Create a key for service account [blue]prefect-cloud-run[/]
+                            - Create a key for service account [blue]{self._service_account_name}[/]
 
                         Updates in Prefect {"workspace" if client.server_type == ServerType.CLOUD else "server"}
 
-                            - Create GCP credentials block [blue]{work_pool_name}-push-pool-credentials[/] to store the service account key
-                    """
+                            - Create GCP credentials block to store the service account key [blue]{self._credentials_block_name}[/]
+                """
             ),
             expand=False,
         )
+
+    async def _customize_resource_names(
+        self, work_pool_name: str, client: PrefectClient
+    ) -> bool:
+        self._service_account_name = prompt(
+            "Please enter a name for the service account",
+            default=self._service_account_name,
+        )
+        self._credentials_block_name = prompt(
+            "Please enter a name for the GCP credentials block",
+            default=self._credentials_block_name,
+        )
+        table = await self._create_provision_table(work_pool_name, client)
+        self._console.print(table)
+
+        return Confirm.ask(
+            "Proceed with infrastructure provisioning?", console=self._console
+        )
+
+    @inject_client
+    async def provision(
+        self,
+        work_pool_name: str,
+        base_job_template: dict,
+        client: Optional[PrefectClient] = None,
+    ) -> Dict[str, Any]:
+        assert client, "Client injection failed"
+        await self._verify_gcloud_ready()
+        self._project = await self._get_project()
+        self._region = await self._get_default_region()
+        self._credentials_block_name = f"{work_pool_name}-push-pool-credentials"
+
+        table = await self._create_provision_table(work_pool_name, client)
         self._console.print(table)
         if self._console.is_interactive:
-            if not Confirm.ask(
-                "Proceed with infrastructure provisioning?", console=self._console
+            chosen_option = prompt_select_from_table(
+                self._console,
+                "Proceed with infrastructure provisioning with default resource names?",
+                [
+                    {"header": "Options:", "key": "option"},
+                ],
+                [
+                    {
+                        "option": (
+                            "Yes, proceed with infrastructure provisioning with default"
+                            " resource names"
+                        )
+                    },
+                    {
+                        "option": (
+                            "Customize names of service account and GCP credentials"
```

Comment:
To be consistent with the prompt "Proceed with infrastructure provisioning with default resource names?" I think we can lump it all under 'resource names'.
```suggestion
                            "Customize resource names?"
```